governator-core/src/main/java/com/netflix/governator/internal/PreDestroyLifecycleActions.java

```diff
@@ -2,55 +2,103 @@
 
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
 import java.util.Collections;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
 
 import javax.annotation.PreDestroy;
 
 import com.netflix.governator.LifecycleAction;
 
 /**
- * Special AbstractLifecycleFeature to support @PreDestroy annotation processing.
- * Note that this feature is implicit in LifecycleModule and therefore does not need 
- * to be added using the LifecycleFeature multibinding.
+ * Special AbstractLifecycleFeature to support @PreDestroy annotation processing and
+ * java.io.AutoCloseable detection. Note that this feature is implicit in LifecycleModule 
+ * and therefore does not need to be added using the LifecycleFeature multibinding.
  * 
  * @author elandau
  */
-public final class PreDestroyLifecycleActions extends AbstractLifecycleFeature {
+public final class PreDestroyLifecycleActions extends AbstractLifecycleFeature<Set<String>> {
```

Comment:
I think forcing all subclasses to work with a 'Set<String>' type for the context, when we can't know exactly how they'll be using the context, is needlessly restrictive with only slight reduction in complexity.  If subclasses are forced to receive a 'Set' or 'Set<?>', then they may as well get a java.lang.Object, as the amount of casting required will be the same to access its content.